pkgs/utils/src/cast.zig

```diff
@@ -0,0 +1,15 @@
+const std = @import("std");
+
+pub fn Cast(To: type, from: anytype) To {
+    var result: To = undefined;
+    inline for (@typeInfo(To).@"struct".fields) |field| {
+        const field_value = switch (@typeInfo(field.type)) {
+            .optional => @field(from, field.name),
+            // TODO: throw error instead of panic?
```

Comment:
that'd be nice because the error can be handled at comptime, when the panic will be called at runtime iirc.